packages/mediapipe-core/lib/src/ffi_utils.dart

```diff
@@ -6,73 +6,131 @@ import 'dart:ffi';
+  /// See also:
+  ///  * [toCharsPointerPointer]
+  Pointer<Char> toNative() => toNativeUtf8().cast<Char>();
+}
 
-/// Converts the C memory representation of a string into a Dart [String]. If the
-/// supplied pointer is a null pointer, this function returns `null`.
-///
-/// See also:
-///  * [toDartStrings]
-String? toDartString(Pointer<Char> val) {
-  if (val.address == 0) return null;
-  return val.cast<Utf8>().toDartString();
+/// Helpers to convert between a `List<String>` and a `Pointer<Pointer<Char>>`.
+extension NativeListOfStrings on List<String> {
+  /// Copies a list of Dart strings into native memory.
+  ///
+  /// See also:
+  ///  * [Pointer<Char>.toNative] for a non-list equivalent.
+  Pointer<Pointer<Char>> toNative() {
+    final ptrArray = calloc<Pointer<Char>>(length);
+    for (var i = 0; i < length; i++) {
+      ptrArray[i] = this[i].toNative();
+    }
+    return ptrArray;
+  }
 }
 
-/// Converts a list of C memory representations of strings into a list of Dart
-/// [String]s.
-///
-/// See also:
-///  * [toDartString]
-List<String?> toDartStrings(Pointer<Pointer<Char>> val, int length) {
-  final dartStrings = <String?>[];
-  int counter = 0;
-  while (counter < length) {
-    dartStrings.add(toDartString(val[counter]));
-    counter++;
+/// Helpers to convert a [Pointer<Char>] into a [Uint8List].
+extension DartAwareChars on Pointer<Char> {
+  /// Creates a Dart view on a pointer to utf8 chars in native memory, cast as
+  /// unsigned, 8-bit integers. This method does not copy the data out of native
+  /// memory.
+  ///
+  /// If this is called on a `nullptr`, the method returns `null`.
+  Uint8List? toUint8List(int length) {
+    if (isNullPointer) return null;
+    RangeError.checkNotNegative(length, 'length');
+    return cast<Uint8>().asTypedList(length);
+  }
+
+  /// Converts the C memory representation of a string into a Dart [String].
+  ///
+  /// If this is called on a `nullptr`, the method returns `null`.
+  ///
+  /// See also:
+  ///  * [toDartStrings]
+  String? toDartString({int? length}) {
+    if (isNullPointer) return null;
+    return cast<Utf8>().toDartString(length: length);
   }
-  return dartStrings;
 }
 
-/// Converts Dart's representation for binary data, a [Uint8List], into its C
-/// memory representation.
-Pointer<Char> prepareUint8List(Uint8List ints) {
-  final Pointer<Uint8> ptr = calloc<Uint8>(ints.length);
-  ptr.asTypedList(ints.length).setAll(0, ints);
-  return ptr.cast();
+/// Helpers to convert between `Pointer<Pointer<Char>>` and `List<String>`.
+extension DartAwarePointerChars on Pointer<Pointer<Char>> {
+  /// Creates a Dart view on a pointer to a pointer to utf8 chars in native memory,
+  /// cast as a `List<String>`. This method does not copy the data out of native
+  /// memory.
+  ///
+  /// If this is called on a `nullptr`, the method returns `null`.
+  ///
+  /// See also:
+  ///  * [toDartString], for a non-list equivalent.
+  List<String?>? toDartStrings(int length) {
```

Comment:
Also here, automatically dealing with nullptrs makes me worried about whether the native memory is freed correctly on the call site of this place. Examples of uses?